aclk/agent_cloud_link.c

```diff
@@ -129,6 +161,31 @@ struct aclk_query_queue {
     u_int64_t count;
 } aclk_queue = { .aclk_query_head = NULL, .aclk_query_tail = NULL, .count = 0 };
 
+/*
+ * After a connection failure -- delay in milliseconds
+ * When a connection is established, the delay function
+ * should be called with mode 0 to reset the fail count
+ */
+unsigned long int aclk_delay(int mode)
+{
+    static int fail = 0;
+
+    if (!mode) {
+        fail = 0;
+        return 0;
+    }
+
+    if (fail == 0) {
+        srandom(time(NULL));
+        fail++;
+        return 0;
+    }
+
+    unsigned long int delay = (unsigned long int)((powl(ACLK_DELAY_SEED, fail)) * 1000 + (random() % 1618));
```

Comment:
This seems to be very over-complicated - will it be significantly better than `1<<fail` ?

Reply:
1 << fail will work (throwing away fibonacci) :+1:  :smile:

Reply:
But, multiply by 2 is completely different of to use the `powl`. 
This is a complete different algorithm for me, I did not enter in the details about the algorithm, because I imagined that there was a motive to use `powl`, why are changing from the fastest function possible to multiply by 2?

Reply:
How is `powl()` faster than bit-shifting for powers of two?

Reply:
It will change to power of two. The delay (for a maximum of 1024) will produce values like

```
0.        0ms
1.     2756ms
2.     4417ms
3.     8346ms
4.    16144ms
5.    32771ms
6.    64401ms
7.   128789ms
8.   256981ms
9.   512283ms
10.  1024ms
```

Reply:
The fibonacci would be
```
0.        0ms
1.     1680ms
2.     2681ms
3.     4960ms
4.     6979ms
5.    11719ms
6.    17963ms
7.    29921ms
8.    46427ms
9.    76081ms
10.   122972ms
11.   198152ms
12.   321754ms
13.   520540ms
14.   842604ms
15.  1024000ms
```

Reply:
> How is `powl()` faster than bit-shifting for powers of two?

It is impossible to be faster.

Reply:
I only suppose that there was a good motive to select Fibonacci instead power of two, but after these answers, it looks like that what really matter is to have a constant to do math.
Finally the functions 2^x and 1.618^x will deliver different outputs, this was the motive I asked more information, because the input is the same, the output never will be. ;)
I do not have any objection to change, I only wanted to understand the motive it was selected 1.618 .